lib/openfoodfacts.dart

```diff
@@ -201,6 +202,31 @@ class OpenFoodAPIClient {
     return result;
   }
 
+  /// Search the OpenFoodFacts product database: multiple barcodes in input.
+  static Future<SearchResult> getProductList(
+    User? user,
+    ProductListQueryConfiguration configuration, {
+    QueryType queryType = QueryType.PROD,
+  }) async {
+    final Uri uri = Uri(
+        scheme: URI_SCHEME,
+        host: queryType == QueryType.PROD ? URI_PROD_HOST : URI_TEST_HOST,
+        path: 'products/${configuration.barcodes.join(',')}.json',
+        queryParameters: configuration.getParametersMap());
+
+    final Response response =
+        await HttpHelper().doGetRequest(uri, user: user, queryType: queryType);
+
+    final String jsonStr = _replaceQuotes(response.body);
+    final SearchResult result = SearchResult.fromJson(json.decode(jsonStr));
+
+    result.products!.asMap().forEach((index, product) {
```

Comment:
We should probably guard against `null` rather than using `!`

Reply:
Fair enough. Initially I just copied the code from other methods. I'll fix all of them.